pkg/validation/internal/multiarch.go

```diff
@@ -0,0 +1,521 @@
+//
+// - If your CSV is missing labels
+//
+// - If your Operator bundle specifies 1images which does not supports all archetypes found (probably supported by your project)
+//
+var MultipleArchitecturesValidator interfaces.Validator = interfaces.ValidatorFunc(multipleArchitecturesValidate)
+
+// ContainerToolsKey defines the key which can be used by its consumers
+// to inform where their container tool that should be used to inspect the image
+const ContainerToolsKey = "container-tools"
+
+// operatorFrameworkArchLabel defines the label used to the store the supported Arch on CSV
+const operatorFrameworkArchLabel = "operatorframework.io/arch."
+
+// operatorFrameworkSoLabel defines the label used to the store the supported So on CSV
+const operatorFrameworkSoLabel = "operatorframework.io/so."
+
+// multiArchValidator store the data to perform the tests
+type multiArchValidator struct {
+	// infraArchLabels store the arch labels (i.e amd64, ppc64le) from
+	// operatorframework.io/arch.<GOARCH>: supported
+	infraArchLabels []string
+	// InfraSOLabel store the OS labels from
+	// operatorframework.io/so.<GOARCH>: supported
+	infraSOLabels []string
+	// images stores the images defined in the bundle with the platform.arch supported
+	images map[string][]platform
+	// allArchFound contains a map of the arch types found
+	allArchFound map[string]string
+	// allOsFound contains a map of the so(s) found
+	allOsFound map[string]string
+	// Store the bundle load
+	bundle *manifests.Bundle
+	// containerTool defines the container tool which will be used to inspect the images
+	containerTool string
+	// warns stores the errors faced by the validator to return the warnings
+	warns []error
+}
+
+// manifestInspect store the data obtained by running container-tool manifest inspect <IMAGE>
+type manifestInspect struct {
+	ManifestData []manifestData `json:"manifests"`
+}
+
+// manifestData store the platforms
+type manifestData struct {
+	Platform platform `json:"platform"`
+}
+
+// platform store the Architecture and OS supported by the image
+type platform struct {
+	Architecture string `json:"architecture"`
+	OS           string `json:"os"`
+}
+
+func multipleArchitecturesValidate(objs ...interface{}) (results []errors.ManifestResult) {
+	// Obtain the k8s version if informed via the objects an optional
+	var containerTool = ""
+	for _, obj := range objs {
+		switch obj.(type) {
+		case map[string]string:
+			// Check the key values informed
+			containerTool = obj.(map[string]string)[ContainerToolsKey]
+			if len(containerTool) > 0 {
+				// Make lower for we compare and use it
+				containerTool = strings.ToLower(containerTool)
+				break
+			}
+		}
+	}
+
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateMultiArchWith(v, containerTool))
```

Comment:
Why are we looping over the objs twice? Can we not do both checks in a single iteration?

Reply:
If we put both together then can we not call the validate before getting the key?
So that is done first to get the key after we call the validate
Note that all are == across the board and I agree that we have a great hall for improvements.
However, wdyt about we let this for when we change one place we do == for all the other places?